### PR TITLE
Notify on logout

### PR DESCRIPTION
**Pull request by yigal-dayan2**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -3,13 +3,15 @@
...
         jcenter()
+//        google()
         maven {
             url 'https://maven.google.com/'
             name 'Google'
         }
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.1.2'
+        classpath "com.android.tools.build:gradle:3.1.1"
```
Comment (rotembr) [inline build.gradle:13]: retrieve back the 3.1.2 v

File: lib/src/main/java/com/ibm/cloud/appid/android/internal/tokenmanager/TokenManager.java [hunk 2]
```diff
@@ -75,6 +77,12 @@
 	private final static String ERROR_DESCRIPTION = "error_description";
 	private final static String ERROR = "error";
 	private final static String INVALID_GRANT = "invalid_grant";
+
+	private static final String OAUTH_ACTIVITY_LOGGING_PATH = "/activity_logging";
+	private static final String ID_KEY = "id_token";
+	private static final String ACTIVITY_KEY = "eventName";
```
Comment (rotembr) [inline lib/src/main/java/com/ibm/cloud/appid/android/internal/tokenmanager/TokenManager.java]: can we name it
EVENT_NAME  and
ID_TOKEN ?

File: lib/src/main/java/com/ibm/cloud/appid/android/internal/tokenmanager/TokenManager.java [hunk 3]
```diff
@@ -352,6 +360,45 @@ protected boolean verifyToken(Key rsaPublicKey, String token, String issuer, Str
...
+			return;
+		}
+
+		String url = Config.getOAuthServerUrl(appId) + OAUTH_ACTIVITY_LOGGING_PATH;
+
+		AppIDRequest request = new AppIDRequest(url, "POST");
+
+		ResponseListener resListener = new ResponseListener() {
+			@Override
+			public void onSuccess(Response response) {
```
Comment (rotembr) [inline lib/src/main/java/com/ibm/cloud/appid/android/internal/tokenmanager/TokenManager.java:388]: add info log on success

File: lib/src/main/java/com/ibm/cloud/appid/android/internal/tokenmanager/TokenManager.java [hunk 4]
```diff
@@ -352,6 +360,45 @@ protected boolean verifyToken(Key rsaPublicKey, String token, String issuer, Str
...
+			}
+			@Override
+			public void onFailure(Response response, Throwable t, JSONObject extendedInfo) {
+				logger.debug("Failed to submit logging request");
```
Comment (rotembr) [inline lib/src/main/java/com/ibm/cloud/appid/android/internal/tokenmanager/TokenManager.java]: should it be debug ? why not error?

File: lib/src/main/java/com/ibm/cloud/appid/android/internal/tokenmanager/TokenManager.java (hunk 2 above)
Comment (yigal-dayan2) [reply lib/src/main/java/com/ibm/cloud/appid/android/internal/tokenmanager/TokenManager.java]: we can't, that's how the json is defined in the spec

File: lib/src/main/java/com/ibm/cloud/appid/android/internal/tokenmanager/TokenManager.java (hunk 3 above)
Comment (yigal-dayan2) [reply lib/src/main/java/com/ibm/cloud/appid/android/internal/tokenmanager/TokenManager.java:388]: already done

File: lib/src/main/java/com/ibm/cloud/appid/android/internal/tokenmanager/TokenManager.java (hunk 4 above)
Comment (yigal-dayan2) [reply lib/src/main/java/com/ibm/cloud/appid/android/internal/tokenmanager/TokenManager.java]: fixed

File: lib/src/main/java/com/ibm/cloud/appid/android/internal/tokenmanager/TokenManager.java (hunk 2 above)
Comment (yigal-dayan2) [reply lib/src/main/java/com/ibm/cloud/appid/android/internal/tokenmanager/TokenManager.java]: changed to EVENT_NAME_KEY, ID_TOKEN_KEY

File: build.gradle (hunk 1 above)
Comment (yigal-dayan2) [reply build.gradle:13]: can you check that v 3.1.2 works for you? It did not work for me and for Vitali. The continuous build also failed.

**Review by rotembr — APPROVED** [review]

**Pull request merged**